packages/design-system/src/components/Box/Box.tsx

```diff
@@ -234,48 +238,53 @@ const Box = forwardRef(<C extends React.ElementType = 'div'>(props: BoxProps<C>,
-  ${({ theme, $paddingRight }) => handleResponsiveValues('padding-inline-end', $paddingRight, theme)}
-  ${({ theme, $paddingBottom }) => handleResponsiveValues('padding-block-end', $paddingBottom, theme)}
-  ${({ theme, $paddingLeft }) => handleResponsiveValues('padding-inline-start', $paddingLeft, theme)}
+    return `1px solid ${theme.colors[$borderColor]}`;
+  }
 
-  ${({ theme, $margin }) => handleResponsiveValues('margin-block', $margin, theme)}
-  ${({ theme, $margin }) => handleResponsiveValues('margin-inline', $margin, theme)}
-  ${({ theme, $marginLeft }) => handleResponsiveValues('margin-inline-start', $marginLeft, theme)}
-  ${({ theme, $marginRight }) => handleResponsiveValues('margin-inline-end', $marginRight, theme)}
-  ${({ theme, $marginTop }) => handleResponsiveValues('margin-block-start', $marginTop, theme)}
-  ${({ theme, $marginBottom }) => handleResponsiveValues('margin-block-end', $marginBottom, theme)}
+  return undefined;
+};
 
-  // Responsive hiding
-  ${({ theme, $hiddenS }) => ($hiddenS ? `${theme.mediaQueries.tablet} { display: none; }` : undefined)}
-  ${({ theme, $hiddenXS }) => ($hiddenXS ? `${theme.mediaQueries.mobile} { display: none; }` : undefined)}
-  
+const generateResponsiveValues = (theme, value) => {
+  if (typeof value === 'undefined') {
+    return undefined;
+  }
 
-  // Borders
-  border-radius: ${({ theme, $hasRadius, $borderRadius }) => ($hasRadius ? theme.borderRadius : $borderRadius)};
-  border-style: ${({ $borderStyle }) => $borderStyle};
-  border-width: ${({ $borderWidth }) => $borderWidth};
-  border-color: ${({ $borderColor, theme }) => extractStyleFromTheme(theme.colors, $borderColor, undefined)};
-  border: ${({ theme, $borderColor, $borderStyle, $borderWidth }) => {
-    // This condition prevents borderColor from override the border-color attribute when not passing borderStyle nor borderWidth
-    if ($borderColor && !$borderStyle && typeof $borderWidth === 'undefined') {
-      return `1px solid ${theme.colors[$borderColor]}`;
-    }
+  if (typeof value === 'object') {
+    const themedValue = Object.keys(value).reduce((acc, key) => {
+      acc[key] = extractStyleFromTheme(theme, value[key], value[key]);
+      return acc;
+    }, {});
 
-    // eslint-disable-next-line consistent-return
-    return undefined;
-  }};
+    return themedValue;
+  }
+  return extractStyleFromTheme(theme, value, value);
+};
+
+const StyledBox = styled.div<PropsToTransientProps<TransientBoxProps>>`
+  ${({ theme, $hiddenM }) =>
+    $hiddenM
+      ? `${theme.breakpoints.medium} { display: none; } ${theme.breakpoints.large} { display: inherit}`
+      : undefined}
+  ${({ theme, $hiddenS }) =>
+    $hiddenS
+      ? `${theme.breakpoints.small} { display: none; } ${theme.breakpoints.medium} { display: inherit}`
+      : undefined}
+  ${({ theme, $hiddenXS }) =>
+    $hiddenXS
+      ? `${theme.breakpoints.initial} { display: none; } ${theme.breakpoints.small} { display: inherit}`
+      : undefined}
```

Comment:
I'd get rid of this and call it a breaking change. People can now pass `display` with a responsive object.